lib/spack/spack/build_systems/oneapi.py

```diff
@@ -57,30 +62,28 @@ def install(self, spec, prefix, installer_path=None):
             raise RuntimeError('install failed')
 
     def setup_run_environment(self, env):
-
         """Adds environment variables to the generated module file.
 
         These environment variables come from running:
 
         .. code-block:: console
 
-           $ source {prefix}/setvars.sh --force
+           $ source {prefix}/{component}/{version}/env/vars.sh
         """
         env.extend(EnvironmentModifications.from_sourcing_file(
-            join(self.prefix, self.component_dir, 'latest/env/vars.sh')))
+            join(self.component_path, 'env', 'vars.sh')))
 
 
 class IntelOneApiLibraryPackage(IntelOneApiPackage):
     """Base class for Intel oneAPI library packages."""
 
     @property
     def headers(self):
-        include_path = '%s/%s/latest/include' % (
-            self.prefix, self.component_dir)
+        include_path = join(self.component_path, 'include')
         return find_headers('*', include_path, recursive=True)
 
     @property
     def libs(self):
-        lib_path = '%s/%s/latest/lib/intel64' % (self.prefix, self.component_dir)
+        lib_path = join(self.component_path, 'lib', 'intel64')
```

Comment:
Why not replace `join` with `join_path` here and elsewhere in this file like you've done in the packages below?

Reply:
See https://spack.readthedocs.io/en/latest/llnl.util.html#module-llnl.util.filesystem for the more "portable" alternatives.

Reply:
Replaced all join with join_path